server/src/main/java/org/opensearch/common/xcontent/XContentHelper.java

```diff
@@ -64,19 +64,12 @@ public static XContentParser createParser(NamedXContentRegistry xContentRegistry
                                               BytesReference bytes) throws IOException {
         Compressor compressor = CompressorFactory.compressor(bytes);
         if (compressor != null) {
-            InputStream compressedInput = null;
-            try {
-                compressedInput = compressor.threadLocalInputStream(bytes.streamInput());
-                if (compressedInput.markSupported() == false) {
-                    compressedInput = new BufferedInputStream(compressedInput);
-                }
-                final XContentType contentType = XContentFactory.xContentType(compressedInput);
-                return XContentFactory.xContent(contentType).createParser(xContentRegistry, deprecationHandler, compressedInput);
-            } catch (Exception e) {
-                if(compressedInput != null)
-                    compressedInput.close();
-                throw e;
+            InputStream compressedInput = compressor.threadLocalInputStream(bytes.streamInput());
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
Analysis of this package determined that this line of code might contain a resource leak. A resource leak can slow down or crash your system.

**Fix**
Consider closing the following resource in a try-finally block: *compressedInput*. Currently, there are execution paths that do not contain closure statements, for example, when XContent.createParser throws an exception.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).